### PR TITLE
Member services

### PR DESCRIPTION
**Pull request by queenieho**

### CODE REVIEW

**Comment by jalehman** [comment]
Hey @queenieho––I changed the base branch that the PR is against (`member-services`) so that we can continue to work against `development` with fixes and minor additions until we're actually ready to launch premium services for members. Please use the `member-services` branch as the base branch for future PRs.

File: src/cljs/iface/components/modal.cljs [hunk 1]
```diff
@@ -0,0 +1,133 @@
+(ns iface.components.modal
```
Comment (jalehman) [inline src/cljs/iface/components/modal.cljs]: A namespace like `iface.components.modal` implies that it contains generic components related to modals; not a *specific modal* like the "Add Service" modal. I'd say that this is a component of *premium services* that happens to be a modal.

File: src/cljs/iface/components/modal.cljs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    {:type     :primary
+     :size     :large
+     :disabled (not can-submit)
+     :on-click on-submit
+     :loading  is-loading}
+    "Add"]])
+
+
+
+(defn service-modal
```
Comment (jalehman) [inline src/cljs/iface/components/modal.cljs]: Arguments passed to a generic component shouldn't need to be derefed; they should just be data.

File: src/cljs/member/services/subs.cljs [hunk 3]
```diff
@@ -109,12 +110,29 @@
    is-visible))
 
 
+(reg-sub
+ :member.services.add-service/required-fields
+ :<- [db/path]
```
Comment (jalehman) [inline src/cljs/member/services/subs.cljs]: What are you using this subscription for?

File: src/cljs/member/services/subs.cljs [hunk 4]
```diff
@@ -109,12 +110,29 @@
...
+ :<- [:member.services.add-service/currently-adding]
+ (fn [[db currently-adding] _]
+   (into [] (filter #(= true (:required %)) (:fields currently-adding)))))
+
+
 (reg-sub
  :member.services.add-service/can-submit?
  :<- [db/path]
```
Comment (jalehman) [inline src/cljs/member/services/subs.cljs]: Looks like the `db/path` subscription is only used to get access to the `:form-data`. Don't we have a dedicated subscription to that?

File: src/cljs/iface/components/modal.cljs (hunk 1 above)
Comment (queenieho) [reply src/cljs/iface/components/modal.cljs]: renaming to iface.components.services

File: src/cljs/member/services/subs.cljs (hunk 3 above)
Comment (queenieho) [reply src/cljs/member/services/subs.cljs]: It's needed in can-submit? subscription. I first tried to incorporate this into a let in can-submit? but for some reason it didn't work.

**Pull request closed without merging**